File: programmers/k진수에서-소수-개수-구하기/s2.js
/**
 * 소수 여부를 판별하는 함수
 * @params {number} n
 * @returns {boolean}
 */
function isPrime(n) {
  if (n < 2) {
    return false;
  }
  for (let i = 2; i <= n ** 0.5; i++) {
    if (n % i === 0) {
      return false;
    }
  }
  return true;
}

function solution(n, k) {
  const kNum = n.toString(k);

  let answer = 0;
  let idx = 0;
  let numStr = "";
  while (idx < kNum.length) {
    if (kNum[idx] === "0" && numStr !== "") {
      if (isPrime(Number(numStr))) {
        answer++;
      }
      numStr = "";
    } else {
      if (kNum[idx] !== "0") {
        numStr += kNum[idx];
      }
    }
    idx++;
  }
  if (isPrime(numStr)) {
    answer++;
  }
  return answer;
}

File: programmers/가장-큰-정사각형-찾기/s2.py
# DP
def solution(board):
    answer = 1 if board[0][0] else 0
    r, c = len(board), len(board[0])
    for i in range(1, r):
        for j in range(1, c):
            if board[i][j]:
                board[i][j] = min(board[i-1][j], board[i][j-1], board[i-1][j-1]) + 1
            if answer < board[i][j]:
                answer = board[i][j]
    return answer ** 2